### PR TITLE
FEATURE: ONE-4035 Pluggable visualization reports drillable measures

### PR DESCRIPTION
**Pull request by ivan-nejezchleb**
All visualizations now reports all possibly drillable items (all measures for now). Used in KD to list such items and user can define drill interactions for such items

---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [ ] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [ ] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [ ] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [ ] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [ ] Successful `extended test - examples`
- [ ] Successful `extended test - storybook`
- [ ] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer:
- gdc-dashboards:

# Related Jira tasks
- ONE-4035: https://jira.intgdc.com/browse/ONE-4035

### CODE REVIEW

**Comment by ivan-nejezchleb** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5156/ : FAILURE in 5m 43s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/590/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by ivan-nejezchleb** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5157/ : SUCCESS in 5m 01s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/592/ : SUCCESS in 7s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2697/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/593/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2133/ : UNSTABLE in 15m 05s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5159/ : SUCCESS in 4m 42s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/595/ : SUCCESS in 24s

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5169/ : SUCCESS in 5m 25s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/607/ : SUCCESS in 8s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5178/ : FAILURE in 1m 25s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/616/ : SUCCESS in 6s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5190/ : SUCCESS in 6m 38s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/639/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5193/ : SUCCESS in 4m 26s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/642/ : SUCCESS in 6s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5194/ : SUCCESS in 5m 32s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/643/ : SUCCESS in 10s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5195/ : FAILURE in 2m 13s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/644/ : SUCCESS in 9s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5196/ : SUCCESS in 8m 59s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/645/ : SUCCESS in 11s

File: src/components/core/base/VisualizationLoadingHOC.tsx [hunk 1]
```diff
@@ -1,9 +1,12 @@
 // (C) 2007-2019 GoodData Corporation
 import * as React from "react";
+
```
Comment (petrjanu) [inline src/components/core/base/VisualizationLoadingHOC.tsx]: ?

File: src/components/core/base/VisualizationLoadingHOC.tsx [hunk 2]
```diff
@@ -200,11 +205,13 @@ export function visualizationLoadingHOC<
                         ...rawExecution,
                         executionResult: executionResultWithResolvedEmptyValues,
                     };
+                    const possibleDrillableItems = this.getPossibleDrillableItems(result.executionResponse);
```
Comment (petrjanu) [inline src/components/core/base/VisualizationLoadingHOC.tsx]: See https://github.com/gooddata/gdc-dashboards/pull/2379/files#r344725393

File: src/components/core/base/VisualizationLoadingHOC.tsx [hunk 3]
```diff
@@ -281,15 +288,42 @@ export function visualizationLoadingHOC<
             }
         };
 
+        private getAllMeasureHeaderItems(
```
Comment (petrjanu) [inline src/components/core/base/VisualizationLoadingHOC.tsx]: Isn't this common-looking selector already present in some service?

File: src/components/core/base/VisualizationLoadingHOC.tsx [hunk 4]
```diff
@@ -281,15 +288,42 @@ export function visualizationLoadingHOC<
...
+                    .map((header: Execution.IMeasureGroupHeader) => header.measureGroupHeader.items),
+            );
+        }
+
+        private getPossibleDrillableItems(response: Execution.IExecutionResponse): IDrillableItemPushData[] {
+            return this.getAllMeasureHeaderItems(response).map(
+                (measure: Execution.IMeasureHeaderItem): IDrillableItemPushData => ({
+                    type: "measure",
+                    localIdentifier: measure.measureHeaderItem.localIdentifier,
+                    title: measure.measureHeaderItem.name,
```
Comment (petrjanu) [inline src/components/core/base/VisualizationLoadingHOC.tsx]: I think measures use both `name` and `title`. In AD/Kd `title` is used as custom name for the measure... Please investigate this. Maybe I'm wrong.

File: src/components/core/base/test/VisualizationLoadingHOC.spec.tsx [hunk 5]
```diff
@@ -305,12 +305,80 @@ describe("VisualizationLoadingHOC", () => {
         });
 
         return testUtils.delay().then(() => {
-            expect(pushData).toHaveBeenCalledWith({
+            expect(pushData).toHaveBeenCalled();
+            expect(pushData.mock.calls[0][0]).toMatchObject({
```
Comment (petrjanu) [inline src/components/core/base/test/VisualizationLoadingHOC.spec.tsx]: Why not `toHaveBeenCalledWith`?

File: stories/internal/PossibleDrillableItems.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
+// (C) 2019 GoodData Corporation
```
Comment (petrjanu) [inline stories/internal/PossibleDrillableItems.tsx]: Add screenshot for this.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5197/ : SUCCESS in 4m 55s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/646/ : SUCCESS in 9s

File: src/components/core/base/test/VisualizationLoadingHOC.spec.tsx (hunk 5 above)
Comment (vlastikcz) [reply src/components/core/base/test/VisualizationLoadingHOC.spec.tsx]: fixed where possible

File: src/components/core/base/VisualizationLoadingHOC.tsx (hunk 3 above)
Comment (ivan-nejezchleb) [reply src/components/core/base/VisualizationLoadingHOC.tsx]: did not find it enywhere

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5202/ : SUCCESS in 9m 00s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/653/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2717/ : UNSTABLE in 20m 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5203/ : SUCCESS in 11m 29s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1271/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/1726/ : SUCCESS in 16m 43s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2719/ : SUCCESS in 12m 30s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5206/ : SUCCESS in 6m 14s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1272/ : SUCCESS in 21s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/1727/ : SUCCESS in 10m 58s

**Pull request merged**